File: src/main/java/com/example/oilrisk_alert/entity/Alert.java
package com.example.oilrisk_alert.entity;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
public class Alert {
    private Long id;
    private LocalDate date;
    private String level;
    private BigDecimal riskIndex;
    private String triggerType;
    private String triggerFactor;
    private String triggerFactorZh;
    private String summary;
    private String summaryEn;
    private String detail;
    private String aiReport;
    private LocalDateTime createdAt;
}

File: frontend/src/types/alert.ts
export type TriggerType = 'THRESHOLD' | 'TREND' | 'ANOMALY'

export interface AlertRecord {
  id: number
  date: string
  level: 'Low' | 'Medium' | 'High'
  riskIndex: number
  triggerType: TriggerType
  triggerFactor: string
  triggerFactorZh: string
  summary: string
  summaryEn: string
  aiReport?: string
}

export interface AlertPage {
  total: number
  page: number
  size: number
  records: AlertRecord[]
}

export interface ApiResponse<T> {
  code: number
  message: string
  data: T
}

export interface TriggerRule {
  ruleType: 'THRESHOLD' | 'TREND' | 'ANOMALY'
  factor: string
  factorZh: string
  currentValue: number
  threshold: number
  description: string
}

export interface AlertDetail extends AlertRecord {
  triggerRules: TriggerRule[]
}
